### ISSUE TITLE
New Icons and Changes for Button Row

### ISSUE THREAD

**Issue by davidamichelson** [srophe/britishLibrary#353]
@wsalesky Could we add two more buttons to the row of buttons with the XML download?

- [x] One button is for e-mailing corrections. Please label it as follows:
Corrections? [Mail Icon]
The mouse over text should say "Click to report a correction via e-mail."
The function should be similar to our other apps. We will want to prefill the subject line of this e-mail with the shelfmark and URI. @wlpotter can you provide Winona with the path for this please? Winona, I think I already have a gmail account set up for SMTP if you need and I have not shared it please let me know.
    - Shelf mark: `/TEI/teiHeader/fileDesc/sourceDesc/msDesc/msIdentifier/altIdentifier/idno[@type="BL-Shelfmark"]`
    - Record URI: `/TEI/teiHeader/fileDesc/sourceDesc/msDesc/msIdentifier/idno[@type="URI"]`

- [x] One button is for auto generating an issue on github.  Please label it as follows:
[GitHub Merge Icon] (no text label) For an icon, what do you or Will suggest? How about something like this: https://github.com/FortAwesome/Font-Awesome/issues/15229? Or if you find something else that would be fine too.
The mouse over text should say "Click to file a data issue on GitHub (requires login)."
We will want to prefill the subject line of this issue with the shelfmark and URI. @wlpotter can you provide Winona with the path for this please and any instructions on how to construct the restful URL for GitHub issues? (Also, how will this work if they are not part of our repo, or do not have a github account, we don't want to open our issue list to robo spam).
    - Shelf mark: `/TEI/teiHeader/fileDesc/sourceDesc/msDesc/msIdentifier/altIdentifier/idno[@type="BL-Shelfmark"]`
    - Record URI: `/TEI/teiHeader/fileDesc/sourceDesc/msDesc/msIdentifier/idno[@type="URI"]`
    - I will add in a below comment the parameters to use for the GitHub issue creation link

As for ordering the buttons I think they should be as follows:

Corrections [mail icon] [GitHub Issue Icon] [Github Code Icon] [TEI] [RDF] [Text] [Print]

I would also like to make the following changes to the existing buttons please

- [x] TEI button: Is there a small enough TEI icon we can use instead? If so let's use that (https://tei-c.org/about/logos/ ?) or https://thenounproject.com/browse/icons/term/tei/ and let's remove the label "TEI/XML" I think the icon is enough.
- [x] RDF button: Is there a small enough RDF icon we can use instead? If so let's use that (https://www.w3.org/2007/10/sw-logos.html ?) or https://www.iconfinder.com/icons/5069289/file_file_extension_file_format_file_type_rdf_icon and let's remove the label "RDF/XML" I think the icon is enough.
- [x] #354

@wlpotter Please feel free to modify what I have suggested here.

**Comment by davidamichelson** [comment]
One more thought, once this issue is done @wsalesky would you open a similar one on the Srophe app main repo? We may want to make these changes across all apps. Thanks!

**Comment by wlpotter** [comment]
A few thoughts here:

- For the GitHub issues, it should be the case that users will need a GitHub account to create issues. I think the data repo allows issues from any user, not just ones we've added as collaborators
- For GitHub issue icons, we could use the pull request one, or maybe the circle with a dot that is the symbol for an open GH issue?
- For the changes to TEI and RDF buttons we need to retain alt text (which could show for most users as mouse-over text) for accessibility reasons, even if we replace the label with an icon. This would apply to the new icons as well

I can look at the xpaths, etc. for constructing the new issue urls when I get a chance today or tomorrow

**Comment by davidamichelson** [comment]
@wlpotter One more question. These html pages are quite long, so once one starts to scroll we lose easy access to these buttons. Should we consider putting them in a sort of menu bar so they scroll?

**Comment by wsalesky** [comment]
@davidamichelson How would you feel about using the mailto function, rather than a popup form? This would force the users local mail client to open and send the email. You loose some control of content but don't need to deal with the SMTP issue.

**Comment by davidamichelson** [comment]
Sure that actually sounds like a great addition across all of our apps! We
can still preformat the subject line with the URI of the record right?

**Comment by wlpotter** [comment]
I have added the xpaths in the [first comment](https://github.com/srophe/britishLibrary/issues/353#issue-2348919653) for the title and shelf mark.

You can specify URL parameters to pre-fill a GitHub issue. We'd probably want something like:

https://github.com/srophe/britishLibrary-data/issues/new?assignees=&labels=community-submitted&title="$SHELFMARK$ | $URI$"

The shelfmark and uri would come from the xpaths noted above.

This will also apply the label "community-submitted", so we can filter by these issues.

**Comment by wsalesky** [comment]
@wlpotter what is the email address for BL?

**Comment by wlpotter** [comment]
Not sure, @davidamichelson ?

**Comment by wsalesky** [comment]
@wlpotter do you like any of these for the 'open a github issue' https://github.com/primer/octicons

Or we could just do a green circle with a dot, and the text 'open'  which is what github does.

**Comment by wsalesky** [comment]
@davidamichelson @wlpotter is it redundant to have a link to the xml in github and a link to the TEI on the server?

**Comment by wsalesky** [comment]
@davidamichelson @wlpotter Still need an email for this as it currently will go to @davidamichelson

**Comment by davidamichelson** [comment]
@wsalesky please see your e-mail for the e-mail address to use for this. Thanks!
